src/nodejs/managed-identity/azure-vm/index.ts

```diff
@@ -30,32 +34,58 @@ const userAssignedClientIdManagedIdentityApplication: ManagedIdentityApplication
     });
 
 const managedIdentityRequestParams: ManagedIdentityRequestParams = {
-    resource: "https://management.azure.com",
+    resource: "https://vault.azure.net",
 };
 
 // self executing anonymous function, needed for async/await usage
 (async () => {
     // system assigned
+    let systemAssignedTokenResponse: AuthenticationResult;
     try {
-        const tokenResponse: AuthenticationResult =
+        systemAssignedTokenResponse =
```

Comment:
What happens when the sample runs without a user assigned client id set in the config? Does it throw and exit? I think we should have a check to see if there is a client id set and only then try to acquire a token using client id. Also, there are options to pass a resource id too. We should have it in the sample too.

Reply:
```
export const userAssignedClientIdConfig: ManagedIdentityConfiguration = {
    system: {
        networkClient,
    },
    managedIdentityIdParams: {
        userAssignedClientId: MANAGED_IDENTITY_RESOURCE_ID,
    },
};
```

if the user were to comment out the `managedIdentityIdParams` block in this config, then it becomes a systemAssigned managed identity. This is normal, expected behavior.

This would be user error if someone tried to define a userAssignedClientId without the actual userAssignedClientId. I'm not really sure what you're getting at here.

Additionally, a resourceId is passed in, in this sample...

Reply:
Let's chat over teams

Reply:
> if the user were to comment out the managedIdentityIdParams block in this config, then it becomes a systemAssigned managed identity. This is normal, expected behavior.

@Robbie-Microsoft - could you please document this in the `README` to make it clear for consuming developers?